### DISCUSSION TITLE
Flashing the image using MacOS

### DISCUSSION THREAD

**Discussion by mvisic** [pgale/lineage_15.1_Installation_TSV#28] [Q&A] [answered]
I managed to get two free TSV's still sealed in a box.

I've installed edl and loaders using the instructions [here](https://github.com/bkerler/edl).

I have then followed the linux instructions in this repository but am stuck on the second start edl command. patch0.xml runs as expected but when I try to run prog_emmc_firehose_8953_ddr.mbn I get **zsh: permission denied**

I then checked permissions for that file using the -ls l command and the file is -rwxrwxrwx (777) Everyone can read, write and execute. 

Is there anything am missing or has anyone successfully flashed a TSV using MacOS?

**Comment by mvisic** [comment, marked as answer]
Had no luck with a Mac, installed Windows 10 on an old Intel NUC to get the job done

**Reply by JGNi** [reply]
What a shame. I must see if I can get hold of a Windows PC then.

**Comment by casdr** [comment]
Flashing using macOS worked fine on my side. These are the commands I used:

```bash
brew install libusb git android-platform-tools
git clone https://github.com/bkerler/edl.git
cd edl
git submodule update --init --recursive
python3 setup.py build
# I had to disable int conversion to fix issues using python3.11
sudo CFLAGS=-Wno-int-conversion python3 setup.py install
```

```bash
export dir=(directory where you placed the firmware files)

edl qfil $dir/image/rawprogram_unsparse.xml $dir/image/patch0.xml $dir/image --loader=$dir/image/prog_emmc_firehose_8953_ddr.mbn

edl w aboot $dir/image/emmc_appsboot.mbn
edl w abootbak $dir/image/emmc_appsboot.mbn

edl w recovery $dir/lineage-15.1-20240531-UNOFFICIAL-starfire-recovery.img

adb sideload $dir/lineage-15.1-20240602-UNOFFICIAL-starfire.zip
adb sideload $dir/open_gapps-arm-8.1-pico-20220215.zip

adb install $dir/WallPanelApp-prod-universal-v0.12.0.apk
```

**Reply by brombram** [reply]
these commands also worked for me, additionally I had to work around an error for missing pylzma, made it work with these steps: https://github.com/bkerler/edl/issues/547#issuecomment-2440149376
